### PR TITLE
add antons json lib

### PR DESCRIPTION
**Pull request by antonsjava**
stream benchmark for antonsjava json library

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -94,6 +94,8 @@ dependencies {
     implementation group: 'com.github.javadev', name: 'underscore', version: '1.88'
     // yasson
     implementation group: 'org.eclipse', name: 'yasson', version: '3.0.2'
+    // antons
+    implementation group: 'io.github.antonsjava', name: 'json', version: '1.17'
```
Comment (fabienrenaud) [inline build.gradle:98]: Libs are sorted in alphabetical order. Please move this to the top, just before avaje.

File: src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java [hunk 2]
```diff
@@ -383,4 +383,177 @@ static User toUser(Value v) {
             return user;
         }
     }
+
+    @Benchmark
+    @Override
+    public Object antons() throws Exception {
+        return antons(sk.antons.json.parse.JsonScanner.instance(JSON_SOURCE().nextString()));
```
Comment (fabienrenaud) [inline src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java:390]: Note that performance wise, most libs tend to be a faster when manipulating bytes/streams as input. So if your lib supports byte/stream input, I suggest you to use that.

File: src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java [hunk 3]
```diff
@@ -165,4 +166,17 @@ public Object purejson() throws Exception {
...
         return Generator.create().generate(purejson);
     }
+
+    @Benchmark
+    @Override
+    public Object antons() throws Exception {
+        sk.antons.json.JsonValue jso = JSON_SOURCE().streamSerializer().antons(JSON_SOURCE().nextPojo());
+
+        ByteArrayOutputStream baos = JsonUtils.byteArrayOutputStream();
+        Writer w = new BufferedWriter(new OutputStreamWriter(baos));
```
Comment (fabienrenaud) [inline src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java:176]: Remove `BufferedWriter`. 
It's useful to minimize I/O when writing to disk, but here the actual output stream writes a byte array which already has the needed capacity. Using `BufferedWriter` here will only contribute to degraded benchmark performance as it introduces an unnecessary intermediate buffer before copying data to the final byte array.

File: src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java [hunk 4]
```diff
@@ -165,4 +166,17 @@ public Object purejson() throws Exception {
...
+        jso.writeCompact(w);
+        w.flush();
```
Comment (fabienrenaud) [inline src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java:178]: Once you've removed the BufferedWriter, you can remove this explicit `flush` call too. `OutputStreamWriter.close()` will auto-flush the buffer.

File: src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java [hunk 5]
```diff
@@ -383,4 +383,177 @@ static User toUser(Value v) {
...
+    }
+
+
+
+    public Users antons(sk.antons.json.parse.JsonScanner scanner) throws IOException {
```
Comment (fabienrenaud) [inline src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java:395]: Please move user deserialization logic to the `UsersStreamDeserializer`

Note: looks like I'll have to move PureJson's code out there too.

**Review by fabienrenaud — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (antonsjava) [reply build.gradle]: done

File: src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java (hunk 2 above)
Comment (antonsjava) [reply src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java]: not in my case, I stay with string

File: src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java (hunk 3 above)
Comment (antonsjava) [reply src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java]: done

File: src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java (hunk 4 above)
Comment (antonsjava) [reply src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java]: done

File: src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java (hunk 5 above)
Comment (antonsjava) [reply src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java]: done

File: src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java [hunk 6]
```diff
@@ -15,6 +15,7 @@
 
 import io.github.senthilganeshs.parser.json.Generator;
 import io.github.senthilganeshs.parser.json.Parser.Value;
+import java.io.BufferedWriter;
```
Comment (fabienrenaud) [inline src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java:18]: not used, remove.

File: src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java [hunk 7]
```diff
@@ -383,4 +383,15 @@ static User toUser(Value v) {
...
+    @Benchmark
+    @Override
+    public Object antons() throws Exception {
+        return JSON_SOURCE().streamDeserializer().antons(JSON_SOURCE().nextString());
+    }
+
+
+
+
+
```
Comment (fabienrenaud) [inline src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java:396]: Extraneous empty lines. Please remove.

**Review by fabienrenaud — CHANGES_REQUESTED** [review]
Please also pot a benchmark result comparing antons serialization/deserialization perf to jackson or some other lib.

File: src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java (hunk 6 above)
Comment (antonsjava) [reply src/main/java/com/github/fabienrenaud/jjb/stream/Serialization.java]: done

File: src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java (hunk 7 above)
Comment (antonsjava) [reply src/main/java/com/github/fabienrenaud/jjb/stream/Deserialization.java]: done

**Comment by antonsjava** [comment]
I start a test for jackson here is log
[antons.log](https://github.com/fabienrenaud/java-json-benchmark/files/11480049/antons.log)

**Comment by fabienrenaud** [comment]
Deserialization results:
```
Benchmark                 Mode  Cnt       Score       Error  Units
Deserialization.antons   thrpt   20  170067.957 ±  6665.711  ops/s
Deserialization.jackson  thrpt   20  281941.501 ± 14398.329  ops/s
```

Serialization results:
```
Benchmark               Mode  Cnt       Score       Error  Units
Serialization.antons   thrpt   20   82924.307 ±  2033.962  ops/s
Serialization.jackson  thrpt   20  279474.988 ± 56244.499  ops/s
```

**Review by fabienrenaud — APPROVED** [review]

**Review by fabienrenaud — APPROVED** [review]

**Comment by fabienrenaud** [comment]
Thank you for your contribution!

**Pull request merged**